acra-censor/common/logging_logic.go

```diff
@@ -181,14 +181,16 @@ func (queryWriter *QueryWriter) readStoredQueries() error {
 		queryWriter.logger.WithError(err).WithField(logging.FieldKeyEventCode, logging.EventCodeErrorCensorIOError).Errorln("Can't read stored queries")
 		return err
 	}
+	queryWriter.mutex.Lock()
 	queryWriter.Queries = q
 	queryWriter.queryIndex = len(q)
+	queryWriter.mutex.Unlock()
```

Comment:
there are several places where we write: `readStoredQueries`, `reset`, `captureQuery`. And mutex protects slice changing `queryWriter.Queries`, not a `queryIndex`. So we can't use only atomics that operates with integers or just pointers.

Reply:
According what I see, there are a lot of potential places for races could be happen, so we need to store multiple atomics Value instead of one mutex, so I think there wont be too much profit in performance.

Reply:
@Lagovas `So we can't use only atomics that operates with integers or just pointers.`
If I am not mistaken, we can use `atomics.Value` as generic atomic Value

Reply:
> If I am not mistaken, we can use atomics.Value as generic atomic Value
we could use it if didn't need to iterate over value and be sure that it will not be changed. in method `captureQuery` we iterate over value to be sure that we don't have such query and then update value. If we will use `sync.Value`, then we load current state of value to iterate. Then we have a gap of time when it may be changed and execution control stole by another goroutine and then save new value with `sync.Value` we new value. But it may be already duplicate because some of background goroutine can append such query.